src/app/components/popup-host/popup-host.component.html

```diff
@@ -0,0 +1,5 @@
+<div class="popup-container" [class.popup-container--hidden]="!popupViewContainer?.length">
```

Comment:
В такой цепочке `!` из-за этого затрудняется чтение. Я бы рекомендовал вынести в геттер данное вычисление, а в шаблоне использовать данный геттер
```suggestion
<div class="popup-container" [class.popup-container--hidden]="shouldShowContainer">
```